### PR TITLE
#2669 Temperature localizing

### LINKED ISSUE #2669: Create Temperature class for localized temperature formatting
## Is your feature request related to a problem? Please describe.

To easily display temperatures as F or C or whatever for a particular language that's been selected, it would be easier to do this similar to how `moment` or `currency` are used. Except for temperatures, there's not really a widely used package

## Describe the solution you'd like

Create a singleton `temperature` which when passed a number returns and object which represents a temperature.

Has methods `format = (scale = "celsius") => `${this.temp}${degree symbol}${celsisus ? C : F}` kind of thing.

## Implementation

I think a global property which sets temperatures to be formatted to celsius or farenheit 

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2669 

## Change summary

- Adds a Temperature class (exported as a singleton function as that seems to be what the big JS packages seem to do and the JS way to do things..! i.e. `moment`, `axios` etc
- Adds use of `temperature` where temps are used.

## Testing

I think the easiest thing to do is add some code to `componentDidMount` in `mSupplyMobileApp.js`!

import: 

`import temperature from './utilities/temperature';`

```
componentDidMount = () => {
    console.log(
      temperature(4)
        .setMetric('f')
        .format()
    );
  };
```

Doesn't lend itself too well to blackbox user testing!

### Related areas to think about

- Probably going to use something at least similar in CCE app - happy for improvement suggestions!
- Not adding 'localizing' currently (kind of the point of the issue?!) as all our customers seemingly use celsius. This was more of a 'be ready for future requirements' kind of thing (also a bit of fun!) as well as going to need it for CCE so why not do it now. To add changing, would add a `setDefaults` function which would just change the default options to use `f` as a default or whatever!
- Wouldv'e preferred to just use a package, but there doesn't seem to be any for this need - i.e. like `currency.js`/`moment` - maybe I should make it :-)
- Surely other metrics aren't really needed.. right?! Kelvin etc?

### CODE REVIEW

**Comment by wlthomson** [comment]
@joshxg I think you can merge directly into `develop` now? 

`v5.1.0` is released and has been merged back into `develop`, so `develop` is now `v6` and `master` is `v5.1`. Would probably keep things simpler, after I made things unnecessarily complex!

**Comment by joshxg** [comment]
@wlthomson cool - just discussing some possible testing with alain at the mo - but will do some mergin' today!

File: src/utilities/temperature.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  format(asMetric) {
+    const { pattern, metric, symbol, negativePattern, formatWithSymbol } = this.options;
+
+    const formatMetric = asMetric || metric;
+    const temperatureValue = this.temperature(formatMetric);
+    const patternToReplace = temperatureValue >= 0 ? pattern : negativePattern;
+
+    if (!Number.isFinite(Number(temperatureValue))) return 'N/A';
```
Comment (mark-prins) [inline src/utilities/temperature.js:63]: this might be overkill: the `N/A` could be either part of the options or set as a const at the top of file.

Comment (joshxg) [reply src/utilities/temperature.js]: nice! Added as an `invalidPattern` option.. maybe a bit strange? But gives more flexibility if ever for some reason needed - 691bf4f

**Pull request merged**